ld/emultempl/elf32.em

```diff
@@ -103,6 +103,7 @@ gld${EMULATION_NAME}_before_parse (void)
 {
   ldfile_set_output_arch ("${OUTPUT_ARCH}", bfd_arch_`echo ${ARCH} | sed -e 's/:.*//'`);
   config.dynamic_link = ${DYNAMIC_LINK-TRUE};
+  config.has_pie = FALSE;
```

Comment:
Hm, it seems a little off that you introduce this config for all targets, but only reset it for 32 bit ELF.  (This binutils will at least be compiled for 64-bit PPU too.)  If the flag is not implicitly cleared (using `memset` or whatever), in which case this line can be removed, I'd say the clearing should also happen in code that is common for all targets.

Reply:
You're absolutely right. i just was too focused on the 32-bit target, namely the SPU here. Need to look into the code whether the flag is cleared implicitly with `memset` or not.
Will apply the necessary change(s).

Reply:
I had a look and there's, unfortunately, no `memset` for the ld_config_type object `config` at a central place, but there's 
`ldmain.c` as the main entry point where a few members of this struct are default initialized. I'm temped to set the default (`FALSE`) for `has_pie` there too. 
ldmain.c:245
` 
    extern int yydebug;
    yydebug = 1;
  }
#endif

  config.build_constructors = TRUE;
  config.rpath_separator = ':';
  config.split_by_reloc = (unsigned) -1;
  config.split_by_file = (bfd_size_type) -1;
  config.make_executable = TRUE;
  config.has_pie = FALSE;
  config.magic_demand_paged = TRUE;
  config.text_read_only = TRUE;

  command_line.warn_mismatch = TRUE;
  command_line.warn_search_mismatch = TRUE;
  command_line.check_section_addresses = -1;
`

What's your opinion on this?

Reply:
Hm, the ideal place would have been the place where `has_shared` is initialized, but it seems this is not done in any central place either.  I think your solution with `ldmain.c` is fine.

Reply:
At least doing it in `ldmain.c` ensures it is initialized to a meaningful value and does not change the "normal" behaviour of `ld`.